File: py/198.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) < 1:
            return 0

        if len(nums) == 1:
            return nums[0]

        res = []
        res.append(nums[0])
        res.append(max(nums[0], nums[1]))

        for i in range(2, len(nums)):
            val = nums[i] + res[i - 2]
            res.append(max(val, res[i - 1]))

        return res[-1]

File: java/198.java
class Solution {
    public int rob(int[] nums) {
        int length = nums.length;
        if(length == 0)
            return 0;
        if(length == 1)
            return nums[0];

        List<Integer> res = new LinkedList<Integer>();

        res.add(nums[0]);
        res.add(Math.max(nums[0], nums[1]));
        int sum;
        for(int i=2; i<length; i++){
            sum = nums[i] + res.get(i-2);
            res.add(Math.max(sum, res.get(i-1)));
        }

        return res.get(length-1);
    }
}